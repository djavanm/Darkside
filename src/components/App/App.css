@font-face {font-family: "STARWARS";
  src: url("//db.onlinewebfonts.com/t/c00548f5d217ed324ae9096051927001.eot");
  src: url("//db.onlinewebfonts.com/t/c00548f5d217ed324ae9096051927001.eot?#iefix") format("embedded-opentype"),
  url("//db.onlinewebfonts.com/t/c00548f5d217ed324ae9096051927001.woff") format("woff"),
  url("//db.onlinewebfonts.com/t/c00548f5d217ed324ae9096051927001.ttf") format("truetype"),
  url("//db.onlinewebfonts.com/t/c00548f5d217ed324ae9096051927001.svg#STARWARS") format("svg");
}

* {
  margin: 0;
}

main {
  display: grid;
  grid-template-columns: 1fr 4fr;
	grid-template-rows: 1fr 5fr;
  grid-template-areas:
    "nav header"
    "nav output";
  height: 100vh;
  background-image: url('http://eskipaper.com/images/star-background-1.jpg');
}

.App-header {
  grid-area: header;
  display: flex;
  justify-content: center;
  align-items: center;
}

h1 {
  font-size: 85px;
  font-family: 'STARWARS';
  color: yellow;
}

h1:hover {
  transform: scale(1.2) perspective(100px);
  transition: all 0.5s ease;
}
