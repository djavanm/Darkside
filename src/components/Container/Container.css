output {
  grid-area: output;
  display: grid;
  grid-template-columns: repeat(4, 1fr);
  grid-gap: 20px;
  padding: 5px 20px;
}

.p_add-favorite {
  font-size: 42px;
  font-family: 'STARWARS';
  color: #FFF;
}

.link {
  text-decoration: none;
  color: black;
}


