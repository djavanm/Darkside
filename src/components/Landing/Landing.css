.Landing_crawl {
  width: 100%;
  background-color: lightblue;
}

.Landing_h2 {
  font-size: 120px;
  margin-bottom: 50px;
  text-align: center;
  text-transform: uppercase;
}

.Landing_p {
  font-size: 60px;
}

.crawler {
  background: transparent !important;
}

.crawler div {
  background: transparent !important;
  min-height: 40vh !important;
}