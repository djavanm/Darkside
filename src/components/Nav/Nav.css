nav {
  grid-area: nav;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: space-around;
  padding: 10px;

}

img {
  height: 130px;
  margin-bottom: 5px;
}

.Nav_div {
  width: 80%;
  min-width: 150px;
  max-width: 200px;
  display: flex;
  flex-direction: column;
  align-items: center;
  border: 3px solid black;
  border-radius: 50%;
  padding: 10px;
  background: lightblue;
}

.Nav_div:hover {
  transform: scale(1.2) perspective(100px);
  transition: all 0.5s ease;
  background: yellow;
}

.Nav_div:hover .favorite-counter {
  color: black;
  text-shadow: none;
}

.Nav_p {
  font-size: 16px;
  font-weight: bold;
  font-family: 'STARWARS';
  padding: 2px;
  color: black;
}

.favorite-counter {
  border: 5px solid black;
  position: absolute;
  top: 42%;
  border-radius: 50%;
  padding: 10px;
  height: 20px;
  width: 20px;
  text-align: center;
  font-size: 18px;
  font-weight: bold;
  color: yellow;
  text-shadow: 1px 1px black;
}

.Nav-favorites {
  position: relative;
}

.active div {
  background-color: yellow;
}