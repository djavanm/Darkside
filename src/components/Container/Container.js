import React from 'react';
import Card from '../Card/Card'
import './Container.css';

const Container = ({ data, type, toggleFavorite, favorites }) => {
  const dataCards = data.map((card, index) => {
     return <Card className='Card-section 'data={card} type={type} key={index} toggleFavorite={toggleFavorite} favorites={favorites} />
  })
  return (
    <output>
      {type === 'favorites' && !favorites.length && <p className="p_add-favorite">Click a card to favorite</p>}
      {dataCards}
    </output>
  )
}

export default Container;