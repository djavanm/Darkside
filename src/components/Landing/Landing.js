import React from 'react';
import Crawl from 'react-star-wars-crawl';
import 'react-star-wars-crawl/lib/index.css'
import './Landing.css';
import PropTypes from 'prop-types';

const Landing = ({ film }) => {
  return (
    <Crawl className="Landing_crawl">
      <h2 className="Landing_h2">{film.title}</h2>
      <p className="Landing_p">{film.body}</p>
    </Crawl>
  );
}

Landing.propTypes = {
  film: PropTypes.object.isRequired,
}
export default Landing;