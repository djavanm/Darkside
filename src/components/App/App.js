import React, { Component } from 'react';
import Nav from '../Nav/Nav';
import Container from '../Container/Container';
import Landing from '../Landing/Landing';
import CardDetails from '../CardDetails/CardDetails';
import apiCalls from '../Util/apiCalls';
import { Route, Link } from 'react-router-dom';
import './App.css';

class App extends Component {
  constructor() {
    super();
    this.state = {
      planets: [],
      people: [],
      vehicles: [],
      favorites: [],
      crawl: {},
      isLoading: true
    }
  }

  toggleFavorite = (card) => {
    const { favorites } = this.state;
    favorites.map(favorite => favorite.name).includes(card.name) ? this.removeFavorite(card) : this.addFavorite(card);
  }

  addFavorite = (card) => {
    const { favorites } = this.state;
    this.setState({ favorites: [...favorites, card]});
  }

  removeFavorite = (card) => {
    const { favorites } = this.state;
    const filteredFavorites = favorites.filter(favorite => favorite.name !== card.name)
    this.setState({ favorites: filteredFavorites});

  }

  componentDidMount() {
    let randomNumber = Math.floor((Math.random() * 7) + 1)
    fetch(`https://swapi.co/api/films/${randomNumber}`)
      .then(res => res.json())
      .then(data => this.setState({ crawl: { title: data.title, body: data.opening_crawl } }))
  

    fetch('https://swapi.co/api/planets/')
      .then(res => res.json())
      .then(data => apiCalls.cleanPlanetData(data.results))
      .then(planets => this.setState({ planets, isLoading: false }))

    fetch('https://swapi.co/api/people/')
      .then(res => res.json())
      .then(data => apiCalls.cleanPeopleData(data.results))
      .then(data => apiCalls.getPeopleHomes(data))
      .then(data => apiCalls.getSpecies(data))
      .then(people => this.setState({ people }))

    fetch('https://swapi.co/api/vehicles/')
      .then(res => res.json())
      .then(data => apiCalls.cleanVehicles(data.results))
      .then(vehicles => this.setState({ vehicles }))
  
    }
    
    render() {
      console.log(this.state)
      const { planets, people, vehicles, favorites, crawl, isLoading } = this.state;
      return (
        <main className="App-main">
        <header className="App-header">
          <Link to='/' className="link">
            <h1>LightSide</h1>
          </Link> 
        </header>
        <Nav favorites={favorites}/>
        <Route exact path='/' render={() => <Landing film={crawl}/> }/>
        <Route exact path='/planets' render={() => <Container data={planets} type="planets" toggleFavorite={this.toggleFavorite} favorites={favorites} />} />
        <Route exact path='/people' render={() => <Container data={people} type="people" toggleFavorite={this.toggleFavorite} favorites={favorites} />} />
        <Route exact path='/vehicles' render={() => <Container data={vehicles} type="vehicles" toggleFavorite={this.toggleFavorite} favorites={favorites} />} />
        <Route exact path='/favorites' render={() => <Container data={favorites} type="favorites" toggleFavorite={this.toggleFavorite} favorites={favorites} />} />
        <Route path='/planets/:name' render={({ match }) => {
            let targetPlanet = this.state.planets.find(planet => planet.name.replace(/\s/g, '') == match.params.name);
            return <CardDetails {...targetPlanet} type="planets" />
        }} />
        <Route path='/people/:name' render={({ match }) => {
            let targetPerson = this.state.people.find(person => person.name.replace(/\s/g, '') == match.params.name);
            return <CardDetails {...targetPerson} type="people" favorites={favorites}/>
        }} />
        <Route path='/vehicles/:name' render={({ match }) => {
            let targetVehicle = this.state.vehicles.find(vehicle => vehicle.name.replace(/\s/g, '').replace("/", "") == match.params.name);
            return <CardDetails {...targetVehicle} type="vehicles" favorites={favorites}/>
        }} />
        <Route path='/favorites/:name' render={({ match }) => {
            let targetFavorite = this.state.favorites.find(favorite => favorite.name.replace(/\s/g, '').replace("/", "") == match.params.name);
            return <CardDetails {...targetFavorite} type="favorites" favorites={favorites}/>
        }} />
      </main>
    );
  }
}

export default App;
