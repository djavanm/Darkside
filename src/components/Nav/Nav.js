import React from 'react';
import './Nav.css';
import { NavLink } from 'react-router-dom';
import PropTypes from 'prop-types'


const Nav = ({favorites, theme}) => {
  const isDark = theme === 'dark' ? 'dark-nav' : '';
  
  return (
    <nav>
      <NavLink to='/people' className="link">
      <div className={`Nav_div Nav-people ${isDark}`} >
        <p className="Nav_p">People</p>
        <img src="https://icon-library.net/images/starwars-icon/starwars-icon-4.jpg" alt="icon of Star Wars droid which displays people cards on click"/>
      </div>
      </NavLink>
      <NavLink to='/planets' className="link">
      <div className={`Nav_div Nav-planets ${isDark}`}>
        <p className="Nav_p">Planets</p>
        <img src="http://icons.iconarchive.com/icons/sensibleworld/starwars/1024/Death-Star-icon.png" alt="death star icon which displays planet cards on click"/>
      </div>
      </NavLink>
      <NavLink to='/vehicles' className="link">
      <div className={`Nav_div Nav-vehicles ${isDark}`}>
        <p className="Nav_p">Vehicles</p>
        <img src="https://cdn2.iconfinder.com/data/icons/star-wars-6/24/Tie-Fighter-512.png" alt="tie fighter icon which displays vehicle cards on click"/>
      </div>
      </NavLink>
      <NavLink to='/favorites' className="link">
      <div className={`Nav_div Nav-favorites ${isDark}`}>
        <p className="Nav_p">Favorites</p>
        <p className={`favorite-counter ${isDark}`}><span>{favorites.length}</span></p>
        <img src="https://upload.wikimedia.org/wikipedia/commons/thumb/b/b4/Linecons_big-star.svg/1024px-Linecons_big-star.svg.png" alt="favorites star which displays favorite on click"/>
      </div>
      </NavLink>
    </nav>
  );
}

Nav.propTypes = {
  favorites: PropTypes.array.isRequired,
  theme: PropTypes.string.isRequired
}

export default Nav;