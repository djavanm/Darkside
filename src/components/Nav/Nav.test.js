import React from 'react';
import { shallow, mount } from 'enzyme';
import App from '../App/App';
import Nav from '../Nav/Nav';
import Landing from '../Landing/Landing';
import Container from '../Container/Container';
import { MemoryRouter } from 'react-router'
import { Route } from 'react-router-dom';

describe('Nav', () => {
  it('should match the snapshot with all data passed in correctly', () => {
    const wrapper = shallow(
      <Nav
        favorites={[]}
        theme="light"
      />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should match the snapshot with all data passed in correctly', () => {
    const wrapper = shallow(
      <Nav
        favorites={[]}
        theme="dark"
      />);
    expect(wrapper).toMatchSnapshot();
  });
});