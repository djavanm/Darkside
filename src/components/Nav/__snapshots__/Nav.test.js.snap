// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav should match the snapshot with all data passed in correctly 1`] = `
<nav>
  <NavLink
    className="link"
    to="/people"
  >
    <div
      className="Nav_div Nav-people "
    >
      <p
        className="Nav_p"
      >
        People
      </p>
      <img
        alt="icon of Star Wars droid which displays people cards on click"
        src="https://icon-library.net/images/starwars-icon/starwars-icon-4.jpg"
      />
    </div>
  </NavLink>
  <NavLink
    className="link"
    to="/planets"
  >
    <div
      className="Nav_div Nav-planets "
    >
      <p
        className="Nav_p"
      >
        Planets
      </p>
      <img
        alt="death star icon which displays planet cards on click"
        src="http://icons.iconarchive.com/icons/sensibleworld/starwars/1024/Death-Star-icon.png"
      />
    </div>
  </NavLink>
  <NavLink
    className="link"
    to="/vehicles"
  >
    <div
      className="Nav_div Nav-vehicles "
    >
      <p
        className="Nav_p"
      >
        Vehicles
      </p>
      <img
        alt="tie fighter icon which displays vehicle cards on click"
        src="https://cdn2.iconfinder.com/data/icons/star-wars-6/24/Tie-Fighter-512.png"
      />
    </div>
  </NavLink>
  <NavLink
    className="link"
    to="/favorites"
  >
    <div
      className="Nav_div Nav-favorites "
    >
      <p
        className="Nav_p"
      >
        Favorites
      </p>
      <p
        className="favorite-counter "
      >
        <span>
          0
        </span>
      </p>
      <img
        alt="favorites star which displays favorite on click"
        src="https://upload.wikimedia.org/wikipedia/commons/thumb/b/b4/Linecons_big-star.svg/1024px-Linecons_big-star.svg.png"
      />
    </div>
  </NavLink>
</nav>
`;

exports[`Nav should match the snapshot with all data passed in correctly 2`] = `
<nav>
  <NavLink
    className="link"
    to="/people"
  >
    <div
      className="Nav_div Nav-people dark-nav"
    >
      <p
        className="Nav_p"
      >
        People
      </p>
      <img
        alt="icon of Star Wars droid which displays people cards on click"
        src="https://icon-library.net/images/starwars-icon/starwars-icon-4.jpg"
      />
    </div>
  </NavLink>
  <NavLink
    className="link"
    to="/planets"
  >
    <div
      className="Nav_div Nav-planets dark-nav"
    >
      <p
        className="Nav_p"
      >
        Planets
      </p>
      <img
        alt="death star icon which displays planet cards on click"
        src="http://icons.iconarchive.com/icons/sensibleworld/starwars/1024/Death-Star-icon.png"
      />
    </div>
  </NavLink>
  <NavLink
    className="link"
    to="/vehicles"
  >
    <div
      className="Nav_div Nav-vehicles dark-nav"
    >
      <p
        className="Nav_p"
      >
        Vehicles
      </p>
      <img
        alt="tie fighter icon which displays vehicle cards on click"
        src="https://cdn2.iconfinder.com/data/icons/star-wars-6/24/Tie-Fighter-512.png"
      />
    </div>
  </NavLink>
  <NavLink
    className="link"
    to="/favorites"
  >
    <div
      className="Nav_div Nav-favorites dark-nav"
    >
      <p
        className="Nav_p"
      >
        Favorites
      </p>
      <p
        className="favorite-counter dark-nav"
      >
        <span>
          0
        </span>
      </p>
      <img
        alt="favorites star which displays favorite on click"
        src="https://upload.wikimedia.org/wikipedia/commons/thumb/b/b4/Linecons_big-star.svg/1024px-Linecons_big-star.svg.png"
      />
    </div>
  </NavLink>
</nav>
`;
